engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -969,10 +1140,10 @@ namespace dmGameSystem
 
         if (sprite_world->m_ReallocBuffers)
         {
-            // Old version has always 4 vertices. New version has up to 8 vertices.
-            // We will allocate for this upper bound
-            uint32_t num_vertices_per_sprite = sprite_world->m_UseGeometries ? 8 : 4;
-            uint32_t num_indices_per_sprite = (num_vertices_per_sprite - 2) * 3;
+            // Since both old and new version supports slice 9, we allocate for this as upper bound
+            // even if all of that space will not be used.
+            uint32_t num_vertices_per_sprite = 16; // 9 slice has 16 points in a grid that we use for the triangles
+            uint32_t num_indices_per_sprite  = 9 * 6; // 9 slice has 9 quads with two triangles each
```

Comment:
It feels excessive tbh.
Suddenly making all sprite data 4x larger seems like something to be cautious about. Also, I'd bet that the slice9 is the rare case when it comes to the sprites.

And, we'll include other attributes in the format soon.

I think it might be time to move to a dmArray for the vertex/index data,
That, or we should move towards a vertex count limit, as opposed to a sprite count limit.

Reply:
Yes, I agree that it's a huge increase in memory especially with more features on the horizon. I think we could keep the sprite count but move the data to arrays as you suggested, I'll fix that